src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.ons.ctp.integration.rhsvc.service.impl;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.List;
+import java.util.stream.Collectors;
+import ma.glasnost.orika.MapperFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.event.model.CollectionCase;
+import uk.gov.ons.ctp.integration.rhsvc.repository.RespondentDataRepository;
+import uk.gov.ons.ctp.integration.rhsvc.representation.CaseDTO;
+import uk.gov.ons.ctp.integration.rhsvc.representation.UniquePropertyReferenceNumber;
+import uk.gov.ons.ctp.integration.rhsvc.service.CaseService;
+
+/** Implementation to deal with Case data */
+@Service
+public class CaseServiceImpl implements CaseService {
+
+  private static final Logger log = LoggerFactory.getLogger(CaseServiceImpl.class);
+
+  @Autowired private RespondentDataRepository dataRepo;
+  @Autowired private MapperFacade mapperFacade;
+
+  private enum CaseType {
```

Comment:
Please make this into an app wide public enum. Even though we made not need it elsewhere, when we do, another will get created and we run the risk of disparity later. Private enum here will get forgotten.

Reply:
Done.